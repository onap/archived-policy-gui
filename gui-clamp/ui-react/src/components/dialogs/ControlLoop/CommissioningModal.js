/*
 * ============LICENSE_START=======================================================
 * Copyright (C) 2021 Nordix Foundation.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 * ============LICENSE_END=========================================================
 */
import Modal from "react-bootstrap/Modal";
import Button from "react-bootstrap/Button";
import React, { useEffect, useState } from "react";
import styled from "styled-components";
import { JSONEditor } from "@json-editor/json-editor";
import ControlLoopService from "../../../api/ControlLoopService";
import OnapConstant from "../../../utils/OnapConstants";

const ModalStyled = styled(Modal)`
  @media (min-width: 800px) {
    .modal-xl {
      max-width: 96%;
    }
  }
  background-color: transparent;
`

const StyledMessagesDiv = styled.div`
  overflow: auto;
  min-width: 100%;
  max-height: 300px;
  padding: 5px 5px 0px 5px;
  text-align: center;
`

const CommissioningModal = (props) => {
  const [windowLocationPathName, setWindowLocationPathName] = useState('');
  const [toscaInitialValues, setToscaInitialValues] = useState({});
  const [toscaJsonSchema, setToscaJsonSchema] = useState({});
  const [jsonEditor, setJsonEditor] = useState({});
  const [show, setShow] = useState(true);

  const handleClose = () => {
    console.log('handleClose called');
    setShow(false);
    props.history.push('/');
  }

  const getToscaServiceTemplateHandler = async (toscaServiceTemplateResponse) => {

    if (!toscaServiceTemplateResponse.ok) {
      const toscaData = await toscaServiceTemplateResponse;
      setToscaInitialValues(toscaData);
    } else {
      const toscaData = await toscaServiceTemplateResponse.json();
      setToscaInitialValues(toscaData);
    }
  }

  const getToscaSchemaHandler = async (toscaSchemaResponse) => {

    if (!toscaSchemaResponse.ok) {
      const toscaSchemaData = await toscaSchemaResponse;
      setToscaJsonSchema(toscaSchemaData);
    } else {
      const toscaSchemaData = await toscaSchemaResponse.json();
      setToscaJsonSchema(toscaSchemaData);
    }
  }

  useEffect(async () => {
    setWindowLocationPathName(window.location.pathname);

    const toscaTemplateResponse = await ControlLoopService.getToscaControlLoopDefinitions(windowLocationPathName)
      .catch(error => error.message);

    const toscaSchemaResponse = await ControlLoopService.getToscaServiceTemplateSchema('node_templates', windowLocationPathName);

    createJsonEditor(await toscaSchemaResponse.json(), await toscaTemplateResponse.json());

    }, []);

  // TODO Need to Hook this up to a new camel endpoint to get it working
  const createJsonEditor = (toscaModel, editorData) => {
    JSONEditor.defaults.options.collapse = true;

    return new JSONEditor(document.getElementById("editor"),
      {
        schema: toscaModel,
        startval: editorData,
        theme: 'bootstrap4',
        iconlib: 'fontawesome5',
        object_layout: 'normal',
        disable_properties: false,
        disable_edit_json: false,
        disable_array_reorder: true,
        disable_array_delete_last_row: true,
        disable_array_delete_all_rows: false,
        array_controls_top: true,
        keep_oneof_values: false,
        collapsed: true,
        show_errors: 'always',
        display_required_only: false,
        show_opt_in: false,
        prompt_before_delete: true,
        required_by_default: false,
      })
  }

  return (
    <ModalStyled size="xl"
                 show={ show }
                 onHide={ handleClose }
                 backdrop="static"
                 keyboard={ false }>
      <Modal.Header closeButton>
        <Modal.Title>Edit Control Loop Parameters</Modal.Title>
      </Modal.Header>
      <br/>
      <div style={ { padding: '5px 5px 0px 5px' } }>
        <Modal.Body>
          <div id="editor"/>
        </Modal.Body>
      </div>
      <Modal.Footer>
        <Button variant="primary"
                >Submit</Button>
        <Button variant="secondary"
                onClick={ handleClose }>Close</Button>
      </Modal.Footer>
    </ModalStyled>
  );
}

export default CommissioningModal;
