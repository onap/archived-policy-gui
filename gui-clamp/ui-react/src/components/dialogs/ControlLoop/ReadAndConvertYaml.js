/*
 * ============LICENSE_START=======================================================
 *  Copyright (C) 2021 Nordix Foundation.
 *  ================================================================================
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 *  SPDX-License-Identifier: Apache-2.0
 *  ============LICENSE_END=========================================================
 */

import React, { useState } from "react";
import GetToscaTemplate from "./GetToscaTemplate";
import Modal from "react-bootstrap/Modal";
import Button from "react-bootstrap/Button";
import { Alert } from "react-bootstrap";

import styled from 'styled-components';
import DeleteToscaTemplate from "./DeleteToscaTemplate";

const ModalStyled = styled(Modal)`
  background-color: transparent;
`

const AlertStyled = styled(Alert)`
  margin-top: 10px;
`

const PreStyled = styled.pre`
  color: #7F0055;
  overflow: auto;
  max-height: 70vh;
  margin-top: 10px;
`

const ReadAndConvertYaml = (props) => {
  const [show, setShow] = useState(true);
  const [toscaTemplateData, setToscaTemplateData] = useState();
  const [deleteToscaTemplateData, setDeleteToscaTemplateData] = useState();
  const [responseOk, setResponseOk] = useState(true);
  const [deleteResponseOk, setDeleteResponseOk] = useState(true);
  const name = 'ToscaServiceTemplateSimple';
  const version = '1.0.0';

  const handleClose = () => {
    console.log('handleClose called');
    setShow(false);
    props.history.push('/');
  }

  const getToscaServiceTemplateHandler = async (toscaServiceTemplateResponse) => {
    // console.log('getToscaServiceTemplateHandler called: ' + toscaServiceTemplate);

    if (!toscaServiceTemplateResponse.ok) {
      console.log('Response is not ok');
      setResponseOk(false);
      const toscaData = await toscaServiceTemplateResponse.text();
      setToscaTemplateData(toscaData);
    } else {
      setResponseOk(true);
      const toscaData = await toscaServiceTemplateResponse.json();
      setToscaTemplateData(toscaData);
    }
  }

  const deleteToscaServiceTemplateHandler = async (deleteToscaServiceTemplateResponse) => {

    if (!deleteToscaServiceTemplateResponse.ok) {
      console.log('Delete response not ok');
      setDeleteResponseOk(false);
      const deleteToscaData = await deleteToscaServiceTemplateResponse;
      setDeleteToscaTemplateData(deleteToscaData);
    } else {
      setDeleteResponseOk(true);
      const deleteToscaData = await deleteToscaServiceTemplateResponse.json();
      setDeleteToscaTemplateData(deleteToscaData);
    }
  }

  return (
    <ModalStyled size="xl"
                 show={ show }
                 onHide={ handleClose }
                 backdrop="static"
                 keyboard={ false }>
      <Modal.Header closeButton>
        <Modal.Title>View Tosca Template</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <GetToscaTemplate templateName={ name }
                          templateVersion={ version }
                          onGetToscaServiceTemplate={ getToscaServiceTemplateHandler }/>
        { responseOk && <PreStyled> { JSON.stringify(toscaTemplateData, null, 2) } </PreStyled> }
        <AlertStyled show={ !responseOk }
                     variant="danger">{ toscaTemplateData }</AlertStyled>
        { deleteResponseOk && responseOk && toscaTemplateData != null &&
        <DeleteToscaTemplate templateName={ name }
                             templateVersion={ version }
                             onDeleteToscaServiceTemplate={ deleteToscaServiceTemplateHandler }
        />
        }
        <AlertStyled show={ !deleteResponseOk }
                     variant="danger">{ deleteToscaTemplateData }</AlertStyled>
        <AlertStyled show={ deleteResponseOk && deleteToscaTemplateData != null }
                     variant="success"><h2>Delete Successful</h2><PreStyled>{ JSON.stringify(deleteToscaTemplateData, null, 2) }</PreStyled></AlertStyled>
      </Modal.Body>
      <Modal.Footer>
        <Button variant="secondary"
                type="null"
                onClick={ handleClose }>Close</Button>
      </Modal.Footer>
    </ModalStyled>
  );
}
;

export default ReadAndConvertYaml;
