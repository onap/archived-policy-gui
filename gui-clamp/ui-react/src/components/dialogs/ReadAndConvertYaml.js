/*-
 * ============LICENSE_START=======================================================
 *  Copyright (C) 20XX-20YY Nordix Foundation.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 * ============LICENSE_END=========================================================
 */
import React, { useState } from "react";
import GetToscaTemplate from "../../api/GetToscaTemplate";
import Modal from "react-bootstrap/Modal";
import Button from "react-bootstrap/Button";

import styled from 'styled-components';

const ModalStyled = styled(Modal)`
  background-color: transparent;
`

const ErrMsgStyled = styled.div`
  color: red;
`

const PreStyled = styled.pre`
  color: #7F0055;
  overflow: auto;
  max-height: 70vh;
`

const ReadAndConvertYaml = (props) => {
  const [show, setShow] = useState(true);
  const [toscaTemplateData, setToscaTemplateData] = useState();
  const name = 'ToscaServiceTemplateSimple';
  const version = '1.0.0';

  const handleClose = () => {
    setShow(false);
    props.history.push('/');
  }

  const getToscaServiceTemplateHandler = (toscaServiceTemplate) => {
    const toscaData = {
      ...toscaServiceTemplate,
      id: Math.random().toString()
    };
    console.log(toscaData);
    setToscaTemplateData(toscaData);
  }

  return (
    <ModalStyled size="xl"
                 show={ show }
                 onHide={ handleClose }
                 backdrop="static"
                 keyboard={ false }>
      <Modal.Header closeButton>
        <Modal.Title>View Tosca Template</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <GetToscaTemplate templateName={ name }
                          templateVersion={ version }
                          onGetToscaServiceTemplate={ getToscaServiceTemplateHandler }/>
        <PreStyled>{ JSON.stringify(toscaTemplateData, null, 2) }</PreStyled>
      </Modal.Body>
      <Modal.Footer>
        <Button variant="secondary"
                type="null"
                onClick={ handleClose }>Cancel</Button>
      </Modal.Footer>
    </ModalStyled>
  );
}

export default ReadAndConvertYaml;
