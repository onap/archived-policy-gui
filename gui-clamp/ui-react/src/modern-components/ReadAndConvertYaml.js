
import React, { useState } from "react";
import GetToscaTemplate from "./GetToscaTemplate";
import Modal from "react-bootstrap/Modal";
import Button from "react-bootstrap/Button";

import styled from 'styled-components';

const ModalStyled = styled(Modal)`
  background-color: transparent;
`

const ErrMsgStyled = styled.div`
  color: red;
`

const PreStyled = styled.pre`
  color: #7F0055;
  overflow: auto;
  max-height: 70vh;
`

const ReadAndConvertYaml = (props) => {
  const [show, setShow] = useState(true);
  const [toscaTemplateData, setToscaTemplateData] = useState();
  const name = 'ToscaServiceTemplateSimple';
  const version = '1.0.0';

  const handleClose = () => {
    setShow(false);
    props.history.push('/');
  }

  const getToscaServiceTemplateHandler = (toscaServiceTemplate) => {
    const toscaData = {
      ...toscaServiceTemplate,
      id: Math.random().toString()
    };
    console.log(toscaData);
    setToscaTemplateData(toscaData);
  }

  return (
    <ModalStyled size="xl"
                 show={ show }
                 onHide={ handleClose }
                 backdrop="static"
                 keyboard={ false }>
      <Modal.Header closeButton>
        <Modal.Title>View Tosca Template</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <GetToscaTemplate templateName={ name }
                          templateVersion={ version }
                          onGetToscaServiceTemplate={ getToscaServiceTemplateHandler }/>
        <PreStyled>{ JSON.stringify(toscaTemplateData, null, 2) }</PreStyled>
      </Modal.Body>
      <Modal.Footer>
        <Button variant="secondary"
                type="null"
                onClick={ handleClose }>Cancel</Button>
      </Modal.Footer>
    </ModalStyled>
  );
}

export default ReadAndConvertYaml;