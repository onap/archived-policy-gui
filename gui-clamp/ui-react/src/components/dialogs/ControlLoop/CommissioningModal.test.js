/*
 * ============LICENSE_START=======================================================
 * Copyright (C) 2021 Nordix Foundation.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 * ============LICENSE_END=========================================================
 */

import React from 'react';
import { mount, shallow } from 'enzyme';
import ReadAndConvertYaml from './ReadAndConvertYaml';
import toJson from "enzyme-to-json";
import { act } from "react-dom/test-utils";
import { createMemoryHistory } from "history";
import CommissioningModal from "./CommissioningModal";

describe('Verify ReadAndConvertYaml', () => {

  it("renders without crashing", () => {
    shallow(<CommissioningModal/>);
  });

  it("renders correctly", () => {
    const tree = shallow(<CommissioningModal/>);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should have three Button elements', () => {
    const container = shallow(<CommissioningModal/>)
    expect(container.find('Button').length).toEqual(3);
  });

  it('handleClose called when bottom button clicked', () => {
    const history = createMemoryHistory();
    const component = mount(<CommissioningModal history={ history }/>)
    const logSpy = jest.spyOn(console, 'log');


    act(() => {
      component.find('[variant="secondary"]').simulate('click');
      expect(logSpy).toHaveBeenCalledWith('handleClose called');
    });
  });

  it('handleClose called when top-right button clicked', () => {
    const history = createMemoryHistory();
    const component = mount(<CommissioningModal history={ history }/>)
    const logSpy = jest.spyOn(console, 'log');


    act(() => {
      component.find('[size="xl"]').get(0).props.onHide();
      expect(logSpy).toHaveBeenCalledWith('handleClose called');
    });
  });

});
