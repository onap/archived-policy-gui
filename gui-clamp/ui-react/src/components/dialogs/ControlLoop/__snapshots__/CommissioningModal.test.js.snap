// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify ReadAndConvertYaml renders correctly 1`] = `
<Styled(Modal)
  backdrop="static"
  keyboard={false}
  onHide={[Function]}
  show={true}
  size="xl"
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle>
      Change Control Loop Common Properties
    </ModalTitle>
  </ModalHeader>
  <br />
  <div
    style={
      Object {
        "padding": "5px 5px 0px 5px",
      }
    }
  >
    <ModalBody>
      <div
        id="editor"
      />
      <Styled(Alert)
        show={false}
        variant="danger"
      >
        Can't get service template:
        <br />
        {}
      </Styled(Alert)>
      <Styled(Alert)
        show={false}
        variant="danger"
      >
        Can't get common properties:
        <br />
        {}
      </Styled(Alert)>
    </ModalBody>
  </div>
  <styled.div />
  <ModalFooter>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      variant="primary"
    >
      Save
    </Button>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      variant="success mr-auto"
    >
      Commission
    </Button>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      variant="secondary"
    >
      Close
    </Button>
  </ModalFooter>
</Styled(Modal)>
`;
